modules/kmm-customizing-upgrades-for-kernel-modules.adoc

```diff
@@ -0,0 +1,63 @@
+// Module included in the following assemblies:
+//
+// * hardware_enablement/kmm-kernel-module-management.adoc
+
+:_content-type: PROCEDURE
+[id="kmm-customizing-upgrades-for-kernel-modules_{context}"]
+= Customizing upgrades for kernel modules
+
+Use this procedure to upgrade the kernel module while running maintenance operations on the node,
+including rebooting the node, if needed. To minimize the impact on the workload running in the cluster, run the upgrade process
+sequentially, one node at a time. Because it requires knowledge of the workload utilizing the kernel module, this process is
+managed by the cluster administrator.
+
+.Prerequisites
+
+* To successfully upgrade a kernel module to a new version, the currently loaded kernel module
+should be unloaded, and a new kernel module file should be loaded.
+This means that you should terminate all user-mode application/workloads using the kernel module
+before attempting to unload the kernel module.
+
+* Ensure that the user workload (the application running in the cluster that is accessing kernel
+module) is not running on the node prior to kernel module unloading and
+that the workload is back running on the node after the new kernel module version has been loaded.
+
+* Ensure that the device plugin managed by KMM on the node is unloaded before the upgrade process
+and reloaded following the upgrade process.
+
+* Due to Kubernetes limitations in label names, the combined length of `Module` name and namespace
+must not exceed 39 characters.
+
+
+.Procedure
+
+. Before upgrading, set the `kmm.node.kubernetes.io/version-module.<module-namespace>.<module-name>=$moduleVersion` label on all the nodes that will be used by the kernel module.
```

Comment:
Provide an `oc` command or an example that applies the label to the specified node.

Reply:
Will address this in the enhancement story.